plugin.py

```diff
@@ -50,3 +52,31 @@ def on_typescript_rename(self, textDocumentPositionParams, respond):
 
         # Server doesn't require any specific response.
         respond(None)
+
+    def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Callable[[], None]) -> bool:
+        if command['command'] == '_typescript.applyCompletionCodeAction':
+            _, items = command['arguments']
+            session = self.weaksession()
+            if session:
+                apply_workspace_edit(session.window, self._to_lsp_edits(items)).then(lambda _: done_callback())
```

Comment:
I was thinking why not use `session.run_code_action_async`, but I believe you get into an infinite
loop in that case...

Reply:
I think we would run into an infinite loop but even if we didn't, this is not a code action. The structure within the arguments is nothing that matches any LSP strcuture.

This is the `command['arguments']`:

```
[
  "c:\\Users\\Predrag\\Documents\\d\\src\\a.js",
  [
    {
      "description": "Import 'foobarbaz' from module \"./comp/b\"",
      "changes": [
        {
          "textChanges": [
            {
              "newText": "const { foobarbaz } = require(\"./comp/b\");\r\n\r\n",
              "end": {
                "line": 1,
                "offset": 1
              },
              "start": {
                "line": 1,
                "offset": 1
              }
            }
          ],
          "fileName": "c:/Users/Predrag/Documents/d/src/a.js"
        }
      ]
    }
  ]
]
```